lib/drivers/medtronic600Driver.js

```diff
@@ -994,6 +999,310 @@ class ReadBasalPatternCommand extends TransmitPacketRequest {
   }
 }
 
+class ReadHistoryInfoResponse extends TransmitPacketResponse {
+  get historySize() {
+    return this.decryptedPayload.readUInt32BE(0x04);
+  }
+
+  get dataStart() {
+    return NGPUtil.NGPTimestamp.fromBuffer(this.decryptedPayload.slice(0x08, 0x10));
+  }
+
+  get dataEnd() {
+    return NGPUtil.NGPTimestamp.fromBuffer(this.decryptedPayload.slice(0x10, 0x18));
+  }
+}
+
+class ReadHistoryInfoCommand extends TransmitPacketRequest {
+  constructor(pumpSession, historyDataType, fromRtc, toRtc) {
+    const params = Buffer.alloc(12);
+    params[0x00] = historyDataType;
+    params[0x01] = 0x04; // Hard coded
+    params.writeUInt32BE(fromRtc, 0x02);
+    params.writeUInt32BE(toRtc, 0x06);
+    params.writeUInt16BE(0x00, 0x0A); // Hard coded
+    super(pumpSession, TransmitPacketRequest.COM_D_COMMAND.READ_HISTORY_INFO_REQUEST,
+      params, ReadHistoryInfoResponse);
+  }
+}
+
+class AckMultipacketCommand extends TransmitPacketRequest {
+  static get SEGMENT_COMMAND() {
+    return {
+      INITIATE_TRANSFER: 0xFF00,
+      SEND_NEXT_SEGMENT: 0xFF01,
+      REPEAT_SEGMENT_PACKETS: 0xFF02,
+    };
+  }
+
+  constructor(pumpSession, segmentCommand) {
+    const params = Buffer.alloc(2);
+    params.writeUInt16BE(segmentCommand);
+    super(pumpSession, TransmitPacketRequest.COM_D_COMMAND.ACK_MULTIPACKET_COMMAND, params);
+  }
+
+  // Override send(), because we don't request an 0x80 response
+  send(hidDevice, readTimeout = BCNLMessage.READ_TIMEOUT_MS, sendDelay = 0, get80response = false) {
+    return super.send(hidDevice, readTimeout, sendDelay, get80response);
+  }
+}
+
+class MultiPacketSegment extends TransmitPacketResponse {
+  get packetNumber() {
+    return this.decryptedPayload.readUInt16BE(0x03);
+  }
+
+  get segmentPayload() {
+    return this.decryptedPayload.slice(0x05, this.decryptedPayload.length - 2);
+  }
+}
+
+// We don't need to inherit from any other message type, because all of the checksums and size
+// checks are done in the MultiPacketSession.
+class ReadHistoryResponse {
+  constructor(dataBlocks) {
+    this.blocks = dataBlocks;
+  }
+
+  get pages() {
+    return _.map(this.blocks, value => value.toString('hex'));
```

Comment:
Nah, don't worry about it. I also stringify my binary blobs for the 5/7-series and store as JSON. If it turns out later that it's slowing things down, we can think more about it then.